### PR TITLE
Small cleanup

### PR DESCRIPTION
**Pull request by unclejack**
**- What I did**
I've cleaned up a bunch of packages found under pkg.

**- How I did it**
I've made the following changes:
- removed comparisons to bool in if conditions
- switched to bytes.Equal where that made sense
- removed for loops where arrays could be appended
- made the code return directly where the returned value was the error return by the called function
**- How to verify it**

**- Description for the changelog**
pkg: clean up a few small issues

### CODE REVIEW

File: pkg/archive/changes.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func (info *FileInfo) addChanges(oldInfo *FileInfo, changes *[]Change) {
 			// breaks down is if some code intentionally hides a change by setting
 			// back mtime
 			if statDifferent(oldStat, newStat) ||
-				bytes.Compare(oldChild.capability, newChild.capability) != 0 {
+				bytes.Equal(oldChild.capability, newChild.capability) {
```
Comment (AkihiroSuda) [inline pkg/archive/changes.go]: logic seems inverted

**Comment by thaJeztah** [comment]
tests are failing, @unclejack (also see @AkihiroSuda's comment)

File: pkg/archive/changes.go (hunk 1 above)
Comment (unclejack) [inline pkg/archive/changes.go]: @AkihiroSuda: Thanks for taking a look and pointing that out. I've fixed it.

**Comment by thaJeztah** [comment]
Interesting;

```
13:41:38 ok  	github.com/docker/docker/pkg/idtools	0.015s	coverage: 17.5% of statements
13:41:40 --- FAIL: TestAtomicWriteToFile (0.00s)
13:41:40 	fswriters_test.go:41: Data mismatch, expected "barbaz", got "barbaz"
13:41:40 --- FAIL: TestAtomicWriteSetCommit (0.00s)
13:41:40 	fswriters_test.go:89: Data mismatch, expected "barbaz", got "barbaz"
13:41:40 FAIL
13:41:40 coverage: 69.5% of statements
```

**Comment by unclejack** [comment]
Everything should be OK now. PTAL

**Comment by stevvooe** [comment]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**